aepsych/benchmark/test_functions.py

```diff
@@ -272,120 +279,121 @@ def new_novel_det(x, scale_factor=1.0):
         Keyword arguments:
-    channel -- 1D array of channel locations whose thresholds to return
+    channel -- 1D tensor of channel locations whose thresholds to return
     scale factor -- scale for the novel_det function, where higher is steeper/lower SD
     wave_freq -- frequency of location waveform on [-1,1]
     target -- target threshold
     """
-    locs = -0.3 * np.sin(5 * wave_freq * (channel - 1 / 6) / np.pi) ** 2 - 0.5
+    locs = -0.3 * torch.sin(5 * wave_freq * (channel - 1 / 6) / torch.pi) ** 2 - 0.5
     scale = (
-        1 / (10 * scale_factor) * (0.75 + 0.25 * np.cos(10 * (0.3 + channel) / np.pi))
+        1 / (10 * scale_factor) * (0.75 + 0.25 * torch.cos(10 * (0.3 + channel) / torch.pi))
     )
     return locs, scale
 
 
-def target_new_novel_det_channels(channel, scale_factor=1.0, wave_freq=1, target=0.75):
+def target_new_novel_det_channels(channel: torch.Tensor, scale_factor: float = 1.0, wave_freq: float = 1, target: float = 0.75) -> torch.Tensor:
     """Get the target (i.e. threshold) for 2D synthetic novel_det(channel) function
         Keyword arguments:
-    channel -- 1D array of channel locations whose thresholds to return
+    channel -- 1D tensor of channel locations whose thresholds to return
     scale factor -- scale for the novel_det function, where higher is steeper/lower SD
     wave_freq -- frequency of location waveform on [-1,1]
     target -- target threshold
     """
-    locs, scale = new_novel_det_channels_params(
-        channel, scale_factor, wave_freq, target
-    )
-    return norm.ppf(target, loc=locs, scale=scale)
+    locs, scale = new_novel_det_channels_params(channel, scale_factor, wave_freq, target)
+    normal_dist = torch.distributions.Normal(locs, scale)
+    return normal_dist.icdf(torch.tensor(target))
 
 
-def new_novel_det_channels(x, channel, scale_factor=1.0, wave_freq=1, target=0.75):
+def new_novel_det_channels(x: torch.Tensor, channel: torch.Tensor, scale_factor: float = 1.0, wave_freq: float = 1, target: float = 0.75) -> torch.Tensor:
     """Get the 2D synthetic novel_det(channel) function
         Keyword arguments:
-    x -- array of shape (n,2) of locations to sample;
+    x -- tensor of shape (n,2) of locations to sample;
          x[...,0] is channel from -1 to 1; x[...,1] is intensity from -1 to 1
     scale factor -- scale for the novel_det function, where higher is steeper/lower SD
     wave_freq -- frequency of location waveform on [-1,1]
     """
-    locs, scale = new_novel_det_channels_params(
-        channel, scale_factor, wave_freq, target
-    )
-    return (x - locs) / scale
+    locs, scale = new_novel_det_channels_params(channel, scale_factor, wave_freq, target)
+    return (x[..., 1] - locs) / scale
 
 
-def cdf_new_novel_det_channels(channel, scale_factor=1.0, wave_freq=1, target=0.75):
+def cdf_new_novel_det_channels(x: torch.Tensor, channel: torch.Tensor, scale_factor: float = 1.0, wave_freq: float = 1, target: float = 0.75) -> torch.Tensor:
```

Comment:
The call signature here is changed (the x variable), why?

Reply:
This was necessary because `cdf_new_novel_det_channels` is calling `new_novel_det_channels` function which requires `x` for the final return `(x - locs) / scale`. Not passing `x` like it was before should cause an error. 
Also, it is documented in the docstring yet not included.

Reply:
Very odd. This function doesn't get used anything (not even in test) so that's why this was never noticed. It's fine to fix it so it is functional.